File: TTObjectClassesForms/Sağlık Lojistik/Eczane Modülleri/Yatan Hasta Reçetesi Modülü/InpatientPrescription/InpatientPrescriptionBaseForm.cs

using System;
using System.Xml;
using System.Data;
using System.Text;
using System.Drawing;
using System.Reflection;
using System.Collections;
using System.Linq;
using System.ComponentModel;
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Collections.ObjectModel;
using System.Runtime.InteropServices;

using TTUtils;
using TTObjectClasses;
using TTDataDictionary;
using TTCoreDefinitions;
using TTConnectionManager;
using TTInstanceManagement;
using TTDefinitionManagement;
using TTStorageManager.Security;

using SmartCardWrapper;

using TTStorageManager;
using System.Runtime.Versioning;
using System.Windows.Forms;
using TTVisual;
namespace TTFormClasses
{
    public partial class InpatientPrescriptionBaseForm : PrescriptionBaseForm
    {
#region InpatientPrescriptionBaseForm_ClientSideMethods
        public void CancelERecete(bool ExistsEHUUOnay)
        {
            
            if (!string.IsNullOrEmpty(_InpatientPrescription.EReceteNo))
            {
                Prescription.InpatientPrescriptionWebCaller callerObject = new Prescription.InpatientPrescriptionWebCaller();
                callerObject.ObjectID = _InpatientPrescription.ObjectID;

                if (string.IsNullOrEmpty(_InpatientPrescription.ERecetePassword))
                    throw new TTException("Doktorun E reçete şifresi girilmemiş.");
                
                if ( _InpatientPrescription.IsSignedPrescription() )
                {
//                    EReceteIslemleri.imzaliEreceteSilCevapDVO response = EReceteIslemleri.WebMethods.imzaliEreceteSilSync(Sites.SiteLocalHost, _InpatientPrescription.ProcedureDoctor.UniqueNo.ToString(), _InpatientPrescription.ERecetePassword.ToString(), PrescriptionBaseForm.GetEReceteSignedDelete(_InpatientPrescription));
//                    if (response != null)
//                    {
//                        if (response.sonucKodu.Equals("0000"))
//                        {
//                            InfoBox.Alert("E reçete başarıyla iptal edilmiştir.");
//                        }
//                        else
//                        {
//                            if (!string.IsNullOrEmpty(response.uyariMesaji))
//                                throw new TTException("e-Reçete Servisinden Gelen Sonuç Mesajı : " + response.sonucMesaji + "\n\r e-Reçete ServisindenGelen Uyarı Mesajı :" + response.uyariMesaji);
//                            else
//                                throw new TTException("e-Reçete Servisinden Gelen Sonuç Mesajı : " + response.sonucMesaji);
//                        }
//                    }
                }
                else
                {
//                    EReceteIslemleri.ereceteSilCevapDVO response = EReceteIslemleri.WebMethods.ereceteSil(Sites.SiteLocalHost, _InpatientPrescription.ProcedureDoctor.UniqueNo.ToString(), _InpatientPrescription.ERecetePassword.ToString(), Prescription.GetEReceteDelete(_InpatientPrescription));
//                    if (response != null)
//                    {
//                        if (response.sonucKodu.Equals("0000"))
//                        {
//                            InfoBox.Alert("E reçete başarıyla iptal edilmiştir.");
//                        }
//                        else
//                        {
//                            if (!string.IsNullOrEmpty(response.uyariMesaji))
//                                throw new TTException("e-Reçete Servisinden Gelen Sonuç Mesajı : " + response.sonucMesaji + "\n\r e-Reçete ServisindenGelen Uyarı Mesajı :" + response.uyariMesaji);
//                            else
//                                throw new TTException("e-Reçete Servisinden Gelen Sonuç Mesajı : " + response.sonucMesaji);
//                        }
//                    }
                }
                if(ExistsEHUUOnay)
                {
                    TTMessage resOnay = EReceteIslemleri.WebMethods.ereceteOnayIptal(Sites.SiteLocalHost, _InpatientPrescription.ProcedureDoctor.UniqueNo.ToString(), _InpatientPrescription.ERecetePassword.ToString(), callerObject, Prescription.GetEreceteApprovalCancelRequest(_InpatientPrescription));
                    if (string.IsNullOrEmpty(_InpatientPrescription.EHURecetePassword) == false && string.IsNullOrEmpty(_InpatientPrescription.EHUUniqueNo) == false)
                    {
                        long uniqueNo = (long)Convert.ToDouble(_InpatientPrescription.EHUUniqueNo);
                        TTMessage ehuOnayIptal = EReceteIslemleri.WebMethods.ereceteOnayIptal(Sites.SiteLocalHost, _InpatientPrescription.EHUUniqueNo, _InpatientPrescription.EHURecetePassword, callerObject, Prescription.GetEreceteEHUCancelRequest(_InpatientPrescription, uniqueNo));
                    }
                } else
                {
                    EReceteIslemleri.WebMethods.ereceteOnayIptal(Sites.SiteLocalHost, _InpatientPrescription.ProcedureDoctor.Person.UniqueRefNo.ToString(), _InpatientPrescription.ERecetePassword.ToString(), callerObject, Prescription.GetEreceteApprovalCancelRequest(_InpatientPrescription));
                }
                
            }
        }

        public void SendSignedEreceteEHUApproval(ResUser currentUser)
        {
            long uniqueNo = (long)Convert.ToDouble(currentUser.UniqueNo);
//            EReceteIslemleri.imzaliEreceteOnayCevapDVO ehuOnay = EReceteIslemleri.WebMethods.imzaliEreceteOnaySync(Sites.SiteLocalHost, currentUser.UniqueNo.ToString(), _InpatientPrescription.ERecetePassword.ToString(), PrescriptionBaseForm.GetEreceteSignedEHUApprovalRequest(_InpatientPrescription, uniqueNo));
//            if (ehuOnay != null)
//            {
//                if (ehuOnay.sonucKodu.Equals("0000"))
//                {
//                    InfoBox.Alert(_InpatientPrescription.EReceteNo.ToString() + " e reçete numarası ile reçete EHU onayı alınmıştır.", MessageIconEnum.InformationMessage);
//                    _InpatientPrescription.EReceteDescription = " E Reçete başarılı bir şekilde EHU onayı almıştır.";
//                    _InpatientPrescription.EHUUniqueNo = currentUser.UniqueNo.ToString();
//
//                    if (_InpatientPrescription.IsRepeated.HasValue && (bool)_InpatientPrescription.IsRepeated.Value)
//                    {
//                        TTObjectContext context = new TTObjectContext(false);
//                        ResUser updatedUser = (ResUser)context.GetObject(currentUser.ObjectID, currentUser.ObjectDef);
//                        updatedUser.ErecetePassword = _InpatientPrescription.EHURecetePassword;
//                        context.Save();
//                        ((ITTObject)currentUser).Refresh();
//                    }
//                }
//                else
//                {
//                    if (!string.IsNullOrEmpty(ehuOnay.uyariMesaji))
//                        throw new TTException("e-Reçete Servisinden Gelen Sonuç Mesajı : " + ehuOnay.sonucMesaji + "\n\r e-Reçete ServisindenGelen Uyarı Mesajı :" + ehuOnay.uyariMesaji);
//                    else
//                        throw new TTException("e-Reçete Servisinden Gelen Sonuç Mesajı : " + ehuOnay.sonucMesaji);
//                }
//            }
        }
        
#endregion InpatientPrescriptionBaseForm_ClientSideMethods
    }
}

File: AtlasWeb/Core/Modules/Saglik_Lojistik/Eczane_Modulleri/Yatan_Hasta_Recetesi_Modulu/InpatientPrescriptionBaseForm.ts
//$612E588E
import { Component, OnInit, NgZone } from '@angular/core';
import { InpatientPrescriptionBaseFormViewModel } from './InpatientPrescriptionBaseFormViewModel';
import { NeHttpService } from 'Fw/Services/NeHttpService';
import { MessageService } from 'Fw/Services/MessageService';
import { Convert } from 'NebulaClient/Mscorlib/Convert';
import { InpatientPrescription } from 'NebulaClient/Model/AtlasClientModel';
import { PrescriptionBaseForm } from "Modules/Saglik_Lojistik/Eczane_Evrensel_Modulu/PrescriptionBaseForm";
import { ResUser } from 'NebulaClient/Model/AtlasClientModel';


@Component({
    selector: 'InpatientPrescriptionBaseForm',
    templateUrl: './InpatientPrescriptionBaseForm.html',
    providers: [MessageService]
})
export class InpatientPrescriptionBaseForm extends PrescriptionBaseForm implements OnInit {
    public inpatientPrescriptionBaseFormViewModel: InpatientPrescriptionBaseFormViewModel = new InpatientPrescriptionBaseFormViewModel();
    public get _InpatientPrescription(): InpatientPrescription {
        return this._TTObject as InpatientPrescription;
    }
    private InpatientPrescriptionBaseForm_DocumentUrl: string = '/api/InpatientPrescriptionService/InpatientPrescriptionBaseForm';
    constructor(protected httpService: NeHttpService,
                protected messageService: MessageService,
                protected ngZone: NgZone) {
        super(httpService, messageService, ngZone);
        this._DocumentServiceUrl = this.InpatientPrescriptionBaseForm_DocumentUrl;
        this.initViewModel();
        this.initFormControls();
    }


    // ***** Method declarations start *****

    public async CancelERecete(ExistsEHUUOnay: boolean): Promise<void> {
      /*  if (!String.isNullOrEmpty(this._InpatientPrescription.EReceteNo)) {
            let callerObject: Prescription.InpatientPrescriptionWebCaller = new Prescription.InpatientPrescriptionWebCaller();
            callerObject.ObjectID = this._InpatientPrescription.ObjectID;
            if (String.isNullOrEmpty(this._InpatientPrescription.ERecetePassword))
                throw new TTException("Doktorun E reçete şifresi girilmemiş.");
            if (this._InpatientPrescription.IsSignedPrescription()) {

            }
            else {

            }
            if (ExistsEHUUOnay) {
                let resOnay: TTMessage = EReceteIslemleri.WebMethods.ereceteOnayIptal(Sites.SiteLocalHost, this._InpatientPrescription.ProcedureDoctor.UniqueNo.toString(), this._InpatientPrescription.ERecetePassword.toString(), callerObject, (await PrescriptionService.GetEreceteApprovalCancelRequest(this._InpatientPrescription)));
                if (String.isNullOrEmpty(this._InpatientPrescription.EHURecetePassword) === false && String.isNullOrEmpty(this._InpatientPrescription.EHUUniqueNo) === false) {
                    let uniqueNo: number = <number>Convert.ToDouble(this._InpatientPrescription.EHUUniqueNo);
                    let ehuOnayIptal: TTMessage = EReceteIslemleri.WebMethods.ereceteOnayIptal(Sites.SiteLocalHost, this._InpatientPrescription.EHUUniqueNo, this._InpatientPrescription.EHURecetePassword, callerObject, (await PrescriptionService.GetEreceteEHUCancelRequest(this._InpatientPrescription, uniqueNo)));
                }
            }
            else {
                EReceteIslemleri.WebMethods.ereceteOnayIptal(Sites.SiteLocalHost, this._InpatientPrescription.ProcedureDoctor.Person.UniqueRefNo.toString(), this._InpatientPrescription.ERecetePassword.toString(), callerObject, (await PrescriptionService.GetEreceteApprovalCancelRequest(this._InpatientPrescription)));
            }
        }*/
    }
    public async SendSignedEreceteEHUApproval(currentUser: ResUser): Promise<void> {
        let uniqueNo: number = <number>Convert.ToDouble(currentUser.UniqueNo);
    }

    // *****Method declarations end *****

    public initViewModel(): void {
        this._TTObject = new InpatientPrescription();
        this.inpatientPrescriptionBaseFormViewModel = new InpatientPrescriptionBaseFormViewModel();
        this._ViewModel = this.inpatientPrescriptionBaseFormViewModel;
        this.inpatientPrescriptionBaseFormViewModel._InpatientPrescription = this._TTObject as InpatientPrescription;
    }

    protected loadViewModel() {
        let that = this;

        that.inpatientPrescriptionBaseFormViewModel = this._ViewModel as InpatientPrescriptionBaseFormViewModel;
        that._TTObject = this.inpatientPrescriptionBaseFormViewModel._InpatientPrescription;
        if (this.inpatientPrescriptionBaseFormViewModel == null)
            this.inpatientPrescriptionBaseFormViewModel = new InpatientPrescriptionBaseFormViewModel();
        if (this.inpatientPrescriptionBaseFormViewModel._InpatientPrescription == null)
            this.inpatientPrescriptionBaseFormViewModel._InpatientPrescription = new InpatientPrescription();

    }

    async ngOnInit()  {
        let that = this;
        await this.load(InpatientPrescriptionBaseFormViewModel);
  
    }




    protected redirectProperties(): void {

    }

    public initFormControls(): void {
        this.Controls = [];

    }


}
